js/src/components/paid-ads/budget-section/budget-recommendation/index.js

```diff
@@ -12,58 +12,75 @@ import useCountryKeyNameMap from '.~/hooks/useCountryKeyNameMap';
 import useFetchBudgetRecommendationEffect from './useFetchBudgetRecommendationEffect';
 import './index.scss';
 
-const BudgetRecommendation = ( props ) => {
-	const { countryCode, dailyAverageCost } = props;
-	const { data: recommendation } = useFetchBudgetRecommendationEffect(
-		countryCode
+/*
+ * If a merchant selects more than one country, the budget recommendation
+ * takes the highest country out from the selected countries.
+ * When looking for the highest one, it should only consider the `daily_budget_high` value.
+ *
+ * For example, a merchant selected Brunei (5-20 USD) and Croatia (10-15 USD),
+ * then the budget recommendation should be (5-20 USD).
+ */
+function getHighestBudget( recommendations ) {
+	return recommendations.reduce( ( defender, challenger ) => {
```

Comment:
💅  It was a bit confusing this `challenger`, `defender` naming  😂 

WDTY of `previous`, `current`?

Reply:
I wonder why `challenger` and `defender` are confusing?

When I use these naming, I thought they were very well placed. I was even a little proud of it. 🙈  Because I thought `defender` could be interpreted intuitively as "the winner of the last round" and `challenger` as "the competitor against the current winner". (Not using `winner` was because it is not yet "winning" when in the loop.)

But maybe it only works in my own background. Perhaps the difference comes from the languages or cultures. In other languages or cultures understanding, they are more easily interpreted to mean something else?

Reply:
After reading the code and understanding well the context, reading comments etc, ofc I understood the metaphor 

Maybe (highest, next)? Anyway is a very minor 💅 issue. 
 
Up to you to change it or not :)